### PR TITLE
Plugin browser feature gate

### PR DESCRIPTION
**Pull request by johnHackworth**
This PR depends on the `FeatureExample` component and the changes introduced at https://github.com/Automattic/wp-calypso/pull/838, so until that PR is merged, this would be unmergeable. Also, the changeset includes the commits on #838, so it will be more easily revieweable once that one is in master.

The plugins browser didn't check at all if the site the user had selected had Manage turned on. Now it does, and if it's off, we show the user the manage error screen, with a `Feature-example` filled with part of the plugin browser content.

image

How to test
=========

1. You need a jetpack site with manage off
2. Go to http://calypso.dev:3000/plugins/browse
3. Select your site
4. You should see the error screen

### CODE REVIEW

**Comment by enejb** [comment]
cc @alternatekev for feedback. 
I think in my conversations with Kevin the idea was to use the Jetpack logo ( in blue ) instead of drake. 

I can't find a mock up from here. To me Drake looks fine :) 

I like the approach of dummy data. Although maybe there is something we can do to make sure that we are always using the latest plugin icon image. I am not 100% sure how easy it would be to implement. Without doing extra calls to .org api. If that is the case we just might have to keep it up to date ourself. 

One thing that is available for users running Jetpack 3.8 are specific attributes to the url that would take the user to a more section specific url. For example if you are to get to the plugins specific section just add 
`&section=plugins` at the end so that the url would be something like
`wp-admin/admin.php?page=jetpack&configure=manage&section=plugins` 
Related https://github.com/Automattic/jetpack/pull/2820

One thing that is missing but could be addressed in a different PR is there are other cases that would could produce a feature gate kind of screen. 
Some that I can think of are:
1. Site doesn't have access to update files. Does it make sense to show the plugin browser in that case?
2. Site is running an older version of Jetpack. Get the user to set the plugin to autoupdate if that is possible or ask the user to update the plugin manually. 
3. Site is not responding. Send them to a place to fix that. 
4. Site has other restrictions to updating files. 

I think it would be great if the feature gate component would take care of those cases as well. 
These case could/should be covered in a different PR. 

One thing that I found missing from this implementation is the Feature gate for the single plugin view.

Which brings me to the question. What do we should when we show to users that have all of their sites with manage disabled?

File: client/my-sites/plugins/main.jsx [hunk 1]
```diff
@@ -633,6 +633,40 @@ export default React.createClass( {
 		return some( this.props.sites.getSelectedOrAllWithPlugins(), site => site && site.jetpack && site.canUpdateFiles );
 	},
 
+	getMockPluginItems: function() {
+		const plugins = [ {
+			slug: 'akismet',
+			name: 'Akismet',
+			wporg: true,
+			icon: '//ps.w.org/akismet/assets/icon-256x256.png'
```
Comment (enejb) [inline client/my-sites/plugins/main.jsx]: I like this. This will hopefully keep the icons up to date! :+1:

**Comment by johnHackworth** [comment]
> One thing that I found missing from this implementation is the Feature gate for the single plugin view.

You got it! :D https://github.com/Automattic/wp-calypso/pull/879

**Comment by johnHackworth** [comment]
Updated with jetpack logo & plugins url parameter:

image

**Comment by johnHackworth** [comment]
> Which brings me to the question. What do we should when we show to users that have all of their sites with manage disabled?

 @enejb do you mean when looking to 'all sites'? I guess we should show this same screen ... But currently I don't think we are checking for that case. They would see all the funcionality, but when they go to a plugin, all their sites would have the "manage disabled" warning. Maybe we just should prevent everybody in that case to access to any plugin-related feature... should we create a new issue?

**Comment by alternatekev** [comment]
@johnHackworth Agreed that this should be a new issue. It won't really work to use this because this is on a per-site basis whereas All Sites obviously could show sites that do in fact have Manage enabled.

**Comment by alternatekev** [comment]
@johnHackworth The margins on this are pretty far off for some reason:

screen shot 2015-12-02 at 9 45 12 am

The margin between the "remote site" text and the "Enable Jetpack Mange" button is 50px but the margin from the button to the "Popular Plugins" section is about 80px. Can we make them both 50px?

**Comment by johnHackworth** [comment]
I've fixed the margins, @alternatekev 

image

They look a little disbalanced yet because we have a bug with plugins browser headers (after the background color change, they kind of dissolve with the background, but they are a independent box:

(these are the borders of that header box)
image

I've created an issue about it:
https://github.com/Automattic/wp-calypso/issues/1226

**Comment by enejb** [comment]
This is ready to ship! Great work! :ship:

**Pull request merged**